.github/workflows/release.yaml

```diff
@@ -0,0 +1,101 @@
+name: "release"
+
+on:
+  push:
+    branches: ["main"]
+
+env:
+  G_DIST_ARTIFACT_NAME: "yscope-log-viewer-dist-${{github.sha}}"
+
+permissions: {}
+
+concurrency:
+  group: "${{github.workflow}}-${{github.ref}}"
+
+  # Cancel in-progress jobs for efficiency
+  cancel-in-progress: true
+
+jobs:
+  build:
+    runs-on: "ubuntu-latest"
+    steps:
+      - uses: "actions/checkout@v4"
+        with:
+          persist-credentials: false
+          submodules: "recursive"
+
+      - uses: "actions/setup-node@v4"
+        with:
+          node-version: "22"
+          cache: "npm"
+          cache-dependency-path: "./package-lock.json"
+
+      - run: "npm install"
```

Comment:
@coderabbitai This should become `npm clean-install` for reproducibility. create an issue to track this